src/System.Linq/src/System/Linq/Select.cs

```diff
@@ -625,13 +635,25 @@ public TResult TryGetLast(out bool found)
                 return sourceFound ? _selector(input) : default(TResult);
             }
 
+            private TResult[] LazyToArray()
```

Comment:
@stephentoub, `ToArray` is generic so it can be regenerated by the JIT. The logic in `LazyToArray` seemed big enough that it seemed worth separating out into a new method, to avoid penalizing code that doesn't go down this branch. I'll post the code size of this method when I get time.